### DISCUSSION TITLE
Calling back the server time will cause the flow limiting middleware of the argo server to reject the request

### DISCUSSION THREAD

**Discussion by yujiangjiang** [argoproj/argo-workflows#12104] [General]
After the argo server is started, the client requests normal access. At this time, the server time is suddenly adjusted for a period of time (for example, 15 minutes). After a period of time, the request to the argo server will return '429 Too Many Requests'. We suspect that a section of code in the current limiting middleware 'github.com/sethvargo/go-limiter' does not support this scenario.
![image](https://github.com/argoproj/argo-workflows/assets/15154648/bcb4fe6f-b53d-489e-a86c-bb63ec8de354)

**Comment by agilgur5** [comment]
> We suspect that a section of code in the current limiting middleware 'github.com/sethvargo/go-limiter' does not support this scenario.

That sounds like you may want to file an issue upstream in [`go-limiter`](https://github.com/sethvargo/go-limiter)? Have you confirmed this?

> At this time, the server time is suddenly adjusted for a period of time (for example, 15 minutes). After a period of time, the request to the argo server will return '429 Too Many Requests'. 

Although this is not really a normal scenario, the server time shouldn't suddenly change. Having a stable server time seems like a sound assumption to me.

In any case, that would be something to discuss upstream.

Also please note that code blocks are [much more accessible](https://meta.stackoverflow.com/a/285557/3431180) than images of code.

**Reply by agilgur5** [reply]
Looks like an upstream issue was filed and received a response: https://github.com/sethvargo/go-limiter/issues/41.

Subsequently a PR was filed and received a review: https://github.com/sethvargo/go-limiter/pull/42

**Reply by yujiangjiang** [reply]
ok
